benchmark-service/benchmark/benchmark.py

```diff
@@ -258,33 +342,62 @@ async def get_commit(app, sha):  # pylint: disable=unused-argument
         assert complete_batch_statuses, batch_statuses
         log.info(f'commit {sha} has a results file')
         status = complete_batch_statuses[-1]
+        batch_id = status['id']
+        log.info(f'status of {sha}: {status}')
     elif running_batch_statuses:
         status = running_batch_statuses[-1]
+        batch_id = status['id']
         log.info(f'batch already exists for commit {sha}')
     else:
         status = None
+        batch_id = None
         log.info(f'no batches or results file exists for {sha}')
 
     commit = {
         'sha': sha,
-        'title': gh_commit['commit']['message'],
+        'title': title,
         'author': gh_commit['commit']['author']['name'],
         'date': gh_commit['commit']['author']['date'],
-        'status': status
+        'status': status,
+        'batch_id': batch_id,
+        'commit_id': commit_id
     }
+    benchmark_data['commits'][sha] = commit
+
+    log.info('got new commits')
+
     return commit
 
 
 async def update_commit(app, sha):  # pylint: disable=unused-argument
+    log.info('in update_commit')
+    global benchmark_data, dates, geo_means, commit_ids
+    gs_reader = app['gs_reader']
     commit = await get_commit(app, sha)
-    if commit['status'] is not None:
-        return commit
+    file_path = f'{BENCHMARK_RESULTS_PATH}/0-{sha}.json'
 
-    batch_client = app['batch_client']
-    batch_id = await submit_test_batch(batch_client, sha)
-    batch = await batch_client.get_batch(batch_id)
-    commit['status'] = batch._last_known_status
-    log.info(f'submitted a batch {batch_id} for commit {sha}')
+    if commit['status'] is None:
+        batch_client = app['batch_client']
+        batch_id = await submit_test_batch(batch_client, sha)
+        batch = await batch_client.get_batch(batch_id)
+        commit['status'] = batch._last_known_status
+        commit['batch_id'] = batch_id
+        log.info(f'submitted a batch {batch_id} for commit {sha}')
+
+    if sha not in benchmark_data['commits']:
+        benchmark_data['commits'][sha] = commit
```

Comment:
This code isn't necessary. `get_commit` already updates the data structure. That code should probably be done above.

```python3
async def update_commit():
    global ...
    commit = await get_commit()
    benchmark_data['commits'][sha] = commit

    if commit['status'] is None:
         ...
         return commit

     has_results_file = ...
     ...
     return commit
```